src/dialog.jsx

```diff
@@ -265,18 +281,18 @@ let Dialog = React.createClass({
     return this.state.open;
   },
 
-  dismiss() {
-    CssEvent.onTransitionEnd(ReactDOM.findDOMNode(this), () => {
-      this.refs.dialogOverlay.allowScrolling();
-    });
+  _testDeprecations() {
+    warning(!this.props.hasOwnProperty('openImmediately'),
```

Comment:
This library wraps the entire body of warning in exactly that statement, we shouldn't have to bother with wrapping these methods as these are already wrapped inside of warning.
See https://github.com/r3dm/warning/blob/master/warning.js